sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -120,7 +120,13 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
       val plan = ShowTables(UnresolvedNamespace(ident), None)
       val ret = sparkSession.sessionState.executePlan(plan).toRdd.collect()
       val tables = ret
-        .map(row => ident ++ Seq(row.getString(1)))
+        .map(row =>
+          // for views, their namespace are empty
```

Comment:
I think it's only true for temp views?

Reply:
`SHOW TABLES` outputs a `isTemporary` column and we can check that directly.

Reply:
I tried to parse that `isTemporary` which is the third string in the row. Neither `row.getString(2).toInt` or `row.getString(2).toBoolean` does not work. I dig into the codebase and seems essentially the RowSerializer uses `Unsafe.putBoolean`.


 Do you know what is the right way to parse a serialized boolean? Should I use a RowDeserializer somehow?

Reply:
For example, for a temp view named `my_temp_table`, this is the internal row in-memory value: `[0,2000000000,200000000d,1,5f706d65745f796d,656c626174]` 

`toCatalystRow(table.namespace().quoted, table.name(), isTempView(table))`. The isTempView is the third value, which is `200000000d`. 

Also I actually don't know why there are five values in the row....